source/signing.c

```diff
@@ -103,7 +108,8 @@ static void s_aws_signing_on_get_credentials(struct aws_credentials *credentials
 
         state->error_code = AWS_AUTH_SIGNING_NO_CREDENTIALS;
     } else {
-        if (state->config.algorithm == AWS_SIGNING_ALGORITHM_V4_ASYMMETRIC) {
+        if (state->config.algorithm == AWS_SIGNING_ALGORITHM_V4_ASYMMETRIC &&
+            !aws_credentials_is_anonymous(credentials)) {
             state->config.credentials = aws_credentials_new_ecc_from_aws_credentials(state->allocator, credentials);
```

Comment:
utterly trivial: put a line-break after the if-statement, now that it spans 2 lines